turbopack/crates/turbopack-ecmascript/src/simple_tree_shake/mod.rs

```diff
@@ -0,0 +1,103 @@
+//! Intermediate tree shaking that uses global information but not good as the full tree shaking.
+
+use anyhow::{bail, Context, Result};
+use rustc_hash::{FxHashMap, FxHashSet};
+use turbo_rcstr::RcStr;
+use turbo_tasks::{ResolvedVc, TryJoinIterExt, Vc};
+use turbopack_core::{
+    module_graph::{ModuleGraph, SingleModuleGraph},
+    resolve::ExportUsage,
+};
+
+use crate::chunk::EcmascriptChunkPlaceable;
+
+#[turbo_tasks::function]
+pub async fn get_module_export_usages(
+    graph: ResolvedVc<ModuleGraph>,
+    module: ResolvedVc<Box<dyn EcmascriptChunkPlaceable>>,
+) -> Result<Vc<ModuleExportUsageInfo>> {
+    let export_usage_info = compute_export_usage_info(graph)
+        .resolve_strongly_consistent()
+        .await?;
+
+    let export_usage_info = export_usage_info.await?;
+
+    let Some(exports) = export_usage_info.used_exports.get(&module) else {
+        bail!(
+            "module not found in export usage info. Something is wrong with the export usage info."
+        );
+    };
+
+    Ok(ModuleExportUsageInfo {
+        exports: exports.clone(),
+    }
+    .cell())
+}
+
+#[turbo_tasks::function(operation)]
+async fn compute_export_usage_info(graph: ResolvedVc<ModuleGraph>) -> Result<Vc<ExportUsageInfo>> {
+    let results = graph
+        .await?
+        .graphs
+        .iter()
+        .map(|g| compute_export_usage_info_single(**g))
+        .try_join()
+        .await?;
+
+    let mut result = ExportUsageInfo::default();
+
+    for item in results {
+        for (k, v) in &item.await?.used_exports {
+            result.used_exports.entry(*k).or_default().extend(v.clone());
+        }
+    }
+
+    Ok(result.cell())
+}
+
+pub async fn compute_export_usage_info_single(
+    graph: Vc<SingleModuleGraph>,
+) -> Result<Vc<ExportUsageInfo>> {
+    let graph = graph.await?;
+    let mut usage = ExportUsageInfo::default();
+
+    // Traverse the module graph
+
+    graph
+        .traverse_edges(|(edge, target)| {
```

Comment:
`traverse_edges` visits only `SingleModuleGraphNode::Module` and not `SingleModuleGraphNode::VisitedModule`

This means that if a module in e.g. the second graph imports a module that was already vistied in the first graph, then that reference will not be visited (and the export will be missing).